Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -571,23 +610,30 @@ private static void EnsureInstallationEnviroment(string installationDir, ITracer
 
             // Try to use given feed
             // If given feed is null, try with feed that from local package
-            // And GetRemoteRepository will fallback to use default feed if pass in feed param is null
-            SourceRepository remoteRepo = GetRemoteRepository(feedUrl ?? localPackageFeedUrl);
+            // And GetRemoteRepositories will fallback to use default feed if pass in feed param is null
+            IEnumerable<SourceRepository> remoteRepos = GetRemoteRepositories(feedUrl ?? localPackageFeedUrl);
 
-            // case 1 and 2
-            if (!string.IsNullOrWhiteSpace(version)
-                && version.Equals(localPackageVersion, StringComparison.OrdinalIgnoreCase)
-                && remoteRepo.PackageSource.Source.Equals(localPackageFeedUrl, StringComparison.OrdinalIgnoreCase))
-            {
-                isInstalled = true;
-            }
-            else if (string.IsNullOrWhiteSpace(version) && string.IsNullOrWhiteSpace(feedUrl))
+            foreach (SourceRepository rr in remoteRepos)
             {
-                // case 3
-                UIPackageMetadata remotePackage = await remoteRepo.GetLatestPackageByIdFromSrcRepo(id);
-                if (remotePackage != null)
+                // case 1 and 2
+                if (!string.IsNullOrWhiteSpace(version)
+                    && version.Equals(localPackageVersion, StringComparison.OrdinalIgnoreCase)
+                    && rr.PackageSource.Source.Equals(localPackageFeedUrl, StringComparison.OrdinalIgnoreCase))
+                {
+                    isInstalled = true;
```

Comment:
This code is actually incorrect when the version is blank, as it prevents upgrading from the first feed to the second feed if you already have the latest from the first (but second has newer one).

Reply:
Actually, maybe the issue is really in `GetLatestPackageByIdFromMetaRes`. When everything in the old feed is unlisted, it shouldn't return anything. But it's still returning the latest. Not entirely sure why.

Reply:
@davidebbo : Fixed.  Submitted a new [pull request](https://github.com/projectkudu/kudu/pull/2568) with the fix.